src/spikeinterface/sortingcomponents/motion/motion_utils.py

```diff
@@ -0,0 +1,566 @@
+        Array with peak locations
+    weight_with_amplitude : bool, default: False
+        If True, motion histogram is weighted by amplitudes
+    avg_in_bin : bool, default True
+        If true, average the amplitudes in each bin.
+        This is done only if weight_with_amplitude=True.
+    direction : "x" | "y" | "z", default: "y"
+        The depth direction
+    bin_s : float, default: 1.0
+        The temporal bin duration in s
+    bin_um : float, default: 2.0
+        The spatial bin size in um. Ignored if spatial_bin_edges is given.
+    hist_margin_um : float, default: 50
+        The margin to add to the minimum and maximum positions before spatial binning.
+        Ignored if spatial_bin_edges is given.
+    spatial_bin_edges : np.array, default: None
+        The pre-computed spatial bin edges
+    depth_smooth_um: None or float
+        Optional gaussian smoother on histogram on depth axis.
+        This is given as the sigma of the gaussian in micrometers.
+    time_smooth_s: None or float
+        Optional gaussian smoother on histogram on time axis.
+        This is given as the sigma of the gaussian in seconds.
+
+    Returns
+    -------
+    motion_histogram
+        2d np.array with motion histogram (num_temporal_bins, num_spatial_bins)
+    temporal_bin_edges
+        1d array with temporal bin edges
+    spatial_bin_edges
+        1d array with spatial bin edges
+    """
+    n_samples = recording.get_num_samples()
+    mint_s = recording.sample_index_to_time(0)
+    maxt_s = recording.sample_index_to_time(n_samples)
+    temporal_bin_edges = np.arange(mint_s, maxt_s + bin_s, bin_s)
+    if spatial_bin_edges is None:
+        spatial_bin_edges = get_spatial_bin_edges(recording, direction, hist_margin_um, bin_um)
+    else:
+        bin_um = spatial_bin_edges[1] - spatial_bin_edges[0]
+
+    arr = np.zeros((peaks.size, 2), dtype="float64")
+    arr[:, 0] = recording.sample_index_to_time(peaks["sample_index"])
+    arr[:, 1] = peak_locations[direction]
+
+    if weight_with_amplitude:
+        weights = np.abs(peaks["amplitude"])
+    else:
+        weights = None
+
+    motion_histogram, edges = np.histogramdd(arr, bins=(temporal_bin_edges, spatial_bin_edges), weights=weights)
+
+    # average amplitude in each bin
+    if weight_with_amplitude and avg_in_bin:
+        bin_counts, _ = np.histogramdd(arr, bins=(temporal_bin_edges, spatial_bin_edges))
+        bin_counts[bin_counts == 0] = 1
+        motion_histogram = motion_histogram / bin_counts
+
+    from scipy.ndimage import gaussian_filter1d
+
+    if depth_smooth_um is not None:
+        motion_histogram = gaussian_filter1d(motion_histogram, depth_smooth_um / bin_um, axis=1, mode="constant")
+
+    if time_smooth_s is not None:
+        motion_histogram = gaussian_filter1d(motion_histogram, time_smooth_s / bin_s, axis=0, mode="constant")
+
+    return motion_histogram, temporal_bin_edges, spatial_bin_edges
+
+
+def make_3d_motion_histograms(
```

Comment:
This is a bit of a pity that make_2d_motion_histogram and makd_3d_motion_histogram are not the same function, one being a particular case of the other. Because conceptually, they should. We could also propose some smoothing for 3d histogram over time, amplitudes and depth. What do you think? One single function could ease the code maintenance

Reply:
let's make it in a follow up PR